### PR TITLE
Cargo holder commands

### PR DESCRIPTION
**Pull request by Mboesn**

### CODE REVIEW

File: src/main/java/frc/robot/commands/EjectCargo.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  private final static double POWER = -0.3;
+  public EjectCargo() {
+    requires(Robot.cargoHolder);
+  }
+
+  // Called repeatedly when this Command is scheduled to run
+  @Override
+  protected void execute() {
+    Robot.cargoHolder.setHolderMotorPower(POWER);
+  }
```
Comment (HilelAmiel) [inline src/main/java/frc/robot/commands/EjectCargo.java:23]: move to init, there is no need to call it repeatedly

File: src/main/java/frc/robot/commands/EjectCargo.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import edu.wpi.first.wpilibj.command.Command;
+import frc.robot.Robot;
+
+/**
+ * this command ejects cargo until the stall stops.
+ */
+public class EjectCargo extends Command {
+  //TODO: calibrate power.
+  private final static double POWER = -0.3;
+  public EjectCargo() {
```
Comment (HilelAmiel) [inline src/main/java/frc/robot/commands/EjectCargo.java:13]: POWER To EJECT_POWER

File: src/main/java/frc/robot/commands/EjectCargo.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  @Override
+  protected void execute() {
+    Robot.cargoHolder.setHolderMotorPower(POWER);
+  }
+
+  // Make this return true when this Command no longer needs to run execute()
+  @Override
+  protected boolean isFinished() {
+    return !Robot.cargoHolder.isCargoCollectedStall();
+  }
```
Comment (HilelAmiel) [inline src/main/java/frc/robot/commands/EjectCargo.java:29]: Robot.cargoHolder.isCargoCollectedStall(); ---> how will it work? I think you should change the command to work by time (for, like, 3 seconds)

File: src/main/java/frc/robot/commands/EjectCargo.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    Robot.cargoHolder.setIsCargoCollected(false);
+  }
+
+  // Called when another command which requires one or more of the same
+  // subsystems is scheduled to run
+  @Override
+  protected void interrupted() {
+    Robot.cargoHolder.setHolderMotorPower(0);
+    Robot.cargoHolder.setIsCargoCollected(true);
+  }
```
Comment (HilelAmiel) [inline src/main/java/frc/robot/commands/EjectCargo.java:44]: Robot.cargoHolder.setIsCargoCollected(true); there is no need. if we push the ball (eject the cargo) we do it with no changes.

**Review by HilelAmiel — CHANGES_REQUESTED** [review]
change

File: src/main/java/frc/robot/commands/EjectCargo.java (hunk 3 above)
Comment (Mboesn) [reply src/main/java/frc/robot/commands/EjectCargo.java:29]: changed to timeout

File: src/main/java/frc/robot/commands/EjectCargo.java (hunk 4 above)
Comment (Mboesn) [reply src/main/java/frc/robot/commands/EjectCargo.java:44]: changed

File: src/main/java/frc/robot/commands/EjectCargo.java (hunk 2 above)
Comment (Mboesn) [reply src/main/java/frc/robot/commands/EjectCargo.java:13]: done

File: src/main/java/frc/robot/commands/CollectCargo.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+ */
+public class CollectCargo extends Command {
+  // TODO: calibrate power.
+  private final static double POWER = 0.3;
+
+  public CollectCargo() {
+    requires(Robot.cargoHolder);
+  }
+
+  // Called repeatedly when this Command is scheduled to run
```
Comment (Mboesn) [inline src/main/java/frc/robot/commands/CollectCargo.java:18]: remove wpi docs

File: src/main/java/frc/robot/commands/Commands.java [hunk 6]
```diff
@@ -0,0 +1,15 @@
+
+package frc.robot.commands;
+
+import edu.wpi.first.wpilibj.command.InstantCommand;
+import frc.robot.Robot;
+
+public class Commands {
```
Comment (Mboesn) [inline src/main/java/frc/robot/commands/Commands.java:7]: add docs

File: src/main/java/frc/robot/commands/EjectCargo.java [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+ */
+public class EjectCargo extends Command {
+  //TODO: calibrate power.
+  private final static double EJECT_POWER = -0.3,TIMEOUT = 2;
+  public EjectCargo() {
+    requires(Robot.cargoHolder);
+    setTimeout(TIMEOUT);
+  }
+
+  // Called repeatedly when this Command is scheduled to run
```
Comment (Mboesn) [inline src/main/java/frc/robot/commands/EjectCargo.java:18]: wpi docs

File: src/main/java/frc/robot/commands/EjectCargo.java [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+  @Override
+  protected void execute() {
+    Robot.cargoHolder.setHolderMotorPower(EJECT_POWER);
+
+  }
+
+  // Make this return true when this Command no longer needs to run execute()
```
Comment (Mboesn) [inline src/main/java/frc/robot/commands/EjectCargo.java:25]: wpi docs

File: src/main/java/frc/robot/commands/EjectCargo.java [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+  @Override
+  protected boolean isFinished() {
+    return isTimedOut();
+  }
+
+  // Called once after isFinished returns true
```
Comment (Mboesn) [inline src/main/java/frc/robot/commands/EjectCargo.java:31]: wpi docs

File: src/main/java/frc/robot/commands/EjectCargo.java [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+  @Override
+  protected void end() {
+    Robot.cargoHolder.setHolderMotorPower(0);
+    Robot.cargoHolder.setIsCargoCollected(false);
+  }
+
+  // Called when another command which requires one or more of the same
+  // subsystems is scheduled to run
```
Comment (Mboesn) [inline src/main/java/frc/robot/commands/EjectCargo.java:39]: wpi docs

File: src/main/java/frc/robot/commands/EjectCargo.java [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+  @Override
+  protected void interrupted() {
+    Robot.cargoHolder.setHolderMotorPower(0);
+    Robot.cargoHolder.setIsCargoCollected(false);
```
Comment (Mboesn) [inline src/main/java/frc/robot/commands/EjectCargo.java:43]: call end

File: src/main/java/frc/robot/commands/KeepCargo.java [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+public class KeepCargo extends Command {
+  // TODO: calibrate power.
+  private final static double STALL_POWER = 0.05;
+  private final static double FULL_POWER = 0.4;
+
+  public KeepCargo() {
+    requires(Robot.cargoHolder);
+  }
+
+  // Called repeatedly when this Command is scheduled to run
```
Comment (Mboesn) [inline src/main/java/frc/robot/commands/KeepCargo.java:19]: wpid docs

File: src/main/java/frc/robot/commands/KeepCargo.java [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+        // keeps the cargo inside.
+        Robot.cargoHolder.setHolderMotorPower(STALL_POWER);
+      } else
+        // recollects the cargo
+        Robot.cargoHolder.setHolderMotorPower(FULL_POWER);
+    } else
+      Robot.cargoHolder.setHolderMotorPower(0);
+  }
+
+  // The command always runs until it is interuppted.
```
Comment (Mboesn) [inline src/main/java/frc/robot/commands/KeepCargo.java:35]: wpid docs

File: src/main/java/frc/robot/commands/KeepCargo.java [hunk 14]
```diff
@@ -0,0 +1,54 @@
...
+  protected void end() {
+    // this shouldn't be called
+    Robot.cargoHolder.setHolderMotorPower(0);
+  }
+
+  // Called when another command which requires one or more of the same
+  // subsystems is scheduled to run
+  @Override
+  protected void interrupted() {
+    Robot.cargoHolder.setHolderMotorPower(0);
```
Comment (Mboesn) [inline src/main/java/frc/robot/commands/KeepCargo.java:52]: end();

**Pull request merged**